### PR TITLE
PHP: Added examples information

### PR DESCRIPTION
**Pull request by VitorVRS**
## Related Issues and Discussions
#520 

## People to notify
@MariagraziaAlastra 

## Testing & Review
To be completed by Language Leader (or DDG Staff)

**Pull Request**
- [x] Title follows correct format (Specifies Instant Answer + Purpose)
- [x] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page**
- [ ] Instant Answer page is correctly filled out and contains:
    - One topic for the Search Space Language (Java, Python, Scala, Ruby, etc.)
    - One topic from: Reference, Help, Libraries, Tools
        - Documentation Fatheads are considered "Reference"
    - Description, Data source, and 2+ example queries
    - Perl Module (e.g. "DDG::Fathead::PerlDoc" -- we only need a name, not an actual file)
    - Source Name (for "More at <source_name>" link)
    - Source Domain (must contain http:// or https:// -- can be the same as Data Source)
    - Source Info (used as Subtitle for each Article -- usually matches the IA Name)
    - 'Skip Abstract' is checked off
    - Source ID (ping @moollaza to assign one, once Fathead is ready for Beta deploy)
- [ ] Fathead Tests are passing (run `$ duckpan test <fathead_id>`)
    - Tester should report any failures

**Code**
- [x] Uniformly indented, well commented
- [x] Fetch.sh and Parse.sh run without errors
- [x] Output contains no blank lines, or multi-line entries

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/php

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
@VitorVRS thanks for this! Please generate the updated output.txt and push it here. Then the Fathead will be automatically installed on Beta! 

You now longer have to wait for staff to deploy it :)

**Comment by VitorVRS** [comment]
Done!

**Comment by gautamkrishnar** [comment]
@VitorVRS Looking good but it is not triggering on beta!
page-shot-2017-1-4 php array_merge at duckduckgo

**Comment by VitorVRS** [comment]
I have no idea what is happening on beta. My last PR wasn't automated like this.

**Comment by gautamkrishnar** [comment]
I had also tested it in codio, its not working `"Sorry, no results were returned from Instant Answer"
1`

**Comment by VitorVRS** [comment]
What can i do?

**Comment by gautamkrishnar** [comment]
@VitorVRS Don't worry. Lets wait for @moollaza.
@mintsoft any idea why this is happening?

**Comment by moollaza** [comment]
@VitorVRS @gautamkrishnar the issue is abstract for `addcslashes` it contains `\0` which creates problems for the DB. You need to escape the slashes there and it will work.

I think a regex like `s/\\(d+)/\\$1/` should work?

**Comment by moollaza** [comment]
Once that's resolved the auto install should work 👍

**Comment by VitorVRS** [comment]
Commited my changes and still don't know what is happening with this autodeploy :joy:

**Comment by moollaza** [comment]
@VitorVRS your fix didn't address the problem in the `cslashes` case. Your regex only looks for `\\x(\d+)` but in this case we don't have any `x`, we just have `\\(\d+)`

Once those slashes are escaped it should install correctly 👍

**Comment by moollaza** [comment]
> Look at this review: #399 (review)

@VitorVRS I don't think `cslashes` was in the output.txt at that point?

> Will fix soon

Thanks!

**Comment by gautamkrishnar** [comment]
@VitorVRS this has some merge conflicts. It  must be fixed... 🔪

**Comment by pastorjamesgardiner50** [comment]
Could you all please take me off this!!

**Comment by moollaza** [comment]
> Could you all please take me off this!!

@pastorjamesgardiner50 unfortunately you need to remove yourself -- You can click the `mute this thread` link in your email, or you will need to click in "unwatch" button at the top if you are subscribed to updates.

**Comment by pjhampton** [comment]
Hey @VitorVRS!

How are you getting on with this? Do you need any help?

**Comment by VitorVRS** [comment]
ping

**Comment by VitorVRS** [comment]
ping

**Comment by gautamkrishnar** [comment]
@VitorVRS really sorry for the late reply... This is good to megrge I think...

**Comment by moollaza** [comment]
@VitorVRS apologies for the delay. It looks like the examples don't contain any newlines? We need to make sure we escape rather than remove them so the code examples are still multiline

File: lib/fathead/php/package.json [hunk 1]
```diff
@@ -6,7 +6,6 @@
   "main": "parse.js",
   "dependencies": {
     "chalk": "^1.1.3",
-    "graceful-fs": "^4.1.9",
```
Comment (moollaza) [inline lib/fathead/php/package.json]: Why is this being removed? It's required to run the parser..

File: lib/fathead/php/parse.js [hunk 2]
```diff
@@ -163,4 +171,68 @@
...
+
+
+        if (!phpcode) {
+            // we have examples section, but without phpcode
+            _log('EXAMPLE WITHOUT CODE: ', key)
+            return content;
+        }
+
+        // get the example description
+        exampleDescription = examples.querySelector('strong');
```
Comment (moollaza) [inline lib/fathead/php/parse.js]: looks like this needs `var` -- I'm seeing an error:

```
[F] Skipping : apc.setup.html
exampleDescription = examples.querySelector('strong');
                           ^
ReferenceError: exampleDescription is not defined
```

**Comment by moollaza** [comment]
@VitorVRS the big issue here is still the missing newlines in the output.txt

If you can get that fixed by tomorrow I can still merge this 👍

File: lib/fathead/php/parse.js (hunk 2 above)
Comment (VitorVRS) [reply lib/fathead/php/parse.js]: fixed

**Pull request closed without merging**